ios/browser/api/sync/brave_sync_worker.cc

```diff
@@ -0,0 +1,430 @@
+  }
+  return sync_code;
+}
+
+bool BraveSyncWorker::IsValidSyncCode(const std::string& sync_code) {
+  std::vector<uint8_t> seed;
+  if (!brave_sync::crypto::PassphraseToBytes32(sync_code, &seed)) {
+    return false;
+  }
+  return seed.size() == SEED_BYTES_COUNT;
+}
+
+bool BraveSyncWorker::SetSyncCode(const std::string& sync_code) {
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
+  if (sync_code.empty()) {
+    return false;
+  }
+
+  auto* sync_service = GetSyncService();
+  if (!sync_service || !sync_service->SetSyncCode(sync_code)) {
+    return false;
+  }
+  return true;
+}
+
+std::string BraveSyncWorker::GetSyncCodeFromHexSeed(const std::string& hex_code_seed) {
+  DCHECK(!hex_code_seed.empty());
+
+  std::vector<uint8_t> bytes;
+		std::string sync_code_words;
+		if (base::HexStringToBytes(hex_code_seed, &bytes)) {
+				DCHECK_EQ(bytes.size(), SEED_BYTES_COUNT);
+				if (bytes.size(), SEED_BYTES_COUNT) {
+						sync_code_words = brave_sync::crypto::PassphraseFromBytes32(bytes);
+						if (sync_code_words.empty()) {
+								VLOG(1) << __func__ << " PassphraseFromBytes32 failed for " << hex_code_seed;
+						}
+				} else {
+						LOG(ERROR) << "wrong seed bytes " << bytes.size();
+				}
+				
+				DCHECK_NE(sync_code_words, "");
+		} else {
+				VLOG(1) << __func__ << " HexStringToBytes failed for " << hex_code_seed;
+		}
+		return sync_code_words;
+}
+
+bool BraveSyncWorker::IsFirstSetupComplete() {
+		syncer::SyncService* sync_service = GetSyncService();
+		return sync_service &&
+										sync_service->GetUserSettings()->IsFirstSetupComplete();
+}
+
+bool BraveSyncWorker::RemoveDevice(const std::string& device_guid) {
+    DCHECK_CURRENTLY_ON(web::WebThread::UI);
+    auto* sync_service =
+        ProfileSyncServiceFactory::GetForBrowserState(browser_state_);
+
+    if (!sync_service || sync_service->GetTransportState() !=
+            syncer::SyncService::TransportState::ACTIVE) {
+      //OnRemoteDeviceInfoDeleted();
+      return false;
+    }
+
+    auto* device_info_service =
+        DeviceInfoSyncServiceFactory::GetForBrowserState(browser_state_);
+    auto* tracker = device_info_service->GetDeviceInfoTracker();
+
+    if (!tracker) {
+      return false;
+    }
+    
+    auto device_list = GetDeviceList();
+    auto device_itr = std::find_if(device_list.begin(),
+                                   device_list.end(),
+                                   [device_guid](const auto& device) {
+                                     return device->guid() == device_guid;
+                                   });
+    
+    if (device_itr == device_list.end()) {
+      return false;
+    }
+    
+    tracker->DeleteDeviceInfo(device_itr->get(),
+                              base::BindOnce(&BraveSyncWorker::OnRemoteDeviceInfoDeleted,
+                                             weak_ptr_factory_.GetWeakPtr()));
+
+    return true;
+}
+
+bool BraveSyncWorker::ResetSync() {
+  DCHECK_CURRENTLY_ON(web::WebThread::UI);
```

Comment:
Now the code around this place is changed in master around this place, and, if there are no iOS-specific obstacles, then it may be would be better to change so not to duplicate the code?

https://github.com/brave/brave-core/blob/f5a69e8d1735f2f724b04e2b5bf666cccc973b50/browser/android/brave_sync_worker.cc#L202

and 

https://github.com/brave/brave-core/blob/f5a69e8d1735f2f724b04e2b5bf666cccc973b50/components/brave_sync/profile_sync_service_helper.cc#L17

Reply:
1.15.x doesn't have such `brave-core/components/brave_sync/profile_sync_service_helper.cc` file. 
Only `master` has it.